src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -125,65 +133,142 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
     setOpenKeys({ ...openKeys, [item.path]: !openKeys[item.path] });
   };
 
+  const handleDownArrow = () => {
+    if (cursor.current + 1 < visibleNodes.current.length) {
+      cursor.current += 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    }
+  };
+
+  const handleUpArrow = () => {
+    if (cursor.current >= 1) {
+      cursor.current -= 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    }
+  };
+
+  const findParent = () => {
+    if (!currentNodeId) return;
+    const parentId = visibleNodes.current.find((el) => el.id === currentNodeId).parent;
+    if (parentId !== '') {
+      cursor.current = visibleNodes.current.findIndex((el) => el.id === parentId);
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    }
+  };
+
+  const findNode = (char) => {
```

Comment:
Can we make these helpers more descriptive in the name?

Reply:
I just updated the names, let me know if they need more work.